### ISSUE TITLE
Play services version conflict with other plugins

### ISSUE THREAD

**Issue by dfibuch** [BranchMetrics/cordova-ionic-phonegap-branch-deep-linking#344]
I've experienced a issue today with building our android app, where it was coming up with an error:

`com.android.build.api.transform.TransformException: java.util.zip.ZipException: duplicate entry: com/google/android/gms/internal/zzed.class`

After some research I pinned this down to one of the plugins we are using requesting a different version of some play-services-xxxx from other plugins in the app. Turns out it was this plugin which recently introduced the use of `build.gradle` and it's using play-services 9.+ and 9+. Looking at other plugins they are all using 11+ now and I believe this has caused the problem where different version of services are being requested.

I've locked the version of this plugin in our app to use 2.6.0 and the app builds again, it was pulling in 2.6.2. I'm not specifically saying there is a bug in this plugin because standalone it will work, but just asking the question whether there is a specific reason for using version 9+ and if I'm right in saying this could be causing the error above or was it something else between 2.6.0 and 2.6.2 that might be the culprit?

**Comment by karlsc** [comment]
Version 2.5.2 is also problematic since today. I tried 2.6.0 like you said and it seems to be building now.

**Comment by ethanneff** [comment]
Hello hello,

@dfibuch can you please send me your output whenever you type `cordova plugins`? I will look at what other plugins are doing and make the following changes to our SDK. Additionally, since you the only one reporting this, I will make an update to 11+ tomorrow and see if this fixes it for you.

@karlsc can you provide me with some errors or behaviors you see so I can investigate? Or did upgrading to `2.6.0` solve your issue?

Thanks

**Comment by karlsc** [comment]
Hi @ethanneff 

We were using this in our config.xml : `<plugin name="branch-cordova-sdk" spec="^2.5.0" />`

And since today, we were not able to build Android.

I tried locking the version up and they all seemed to work. Then I tested the latest version `2.6.2` and realized that this was the culprit.

Here are the logs and the error:

`:transformClassesWithJarMergingForDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformClassesWithJarMergingForDebug'.
> com.android.build.api.transform.TransformException: java.util.zip.ZipException: duplicate entry: com/google/android/gms/internal/zzcq.class

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 4.342 secs
Error: /project/gradlew: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Warning: AndroidManifest.xml already defines debuggable (in http://schemas.android.com/apk/res/android); using existing value in manifest.

Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformClassesWithJarMergingForDebug'.
> com.android.build.api.transform.TransformException: java.util.zip.ZipException: duplicate entry: com/google/android/gms/internal/zzcq.class`

Hope this helps!

**Comment by ethanneff** [comment]
@karlsc Can you also give me the output for `cordova plugins`? So I can see what our SDK is conflicting with?

Thanks,

**Comment by dfibuch** [comment]
@ethanneff Here is the output you requested:
![image](https://user-images.githubusercontent.com/4235431/28163132-b7058fb0-67c1-11e7-9ffc-be67a166f118.png)

You might also want to consider making this at least a minor if not a major release because it's introducing breaking changes. We had our config use ~2.6.0 so it picked up the patch version, making it a minor version change would have not worked for @karlsc because he had it as ^2.5.0. Just a thought.

With the version 11+, I spent half a day of fun and games with plugins using various versions and noticed they were all updating their code to 11+ (See: https://github.com/phonegap/phonegap-plugin-push/commit/9de2707880b17e0e4039ba05ee5d7b9a435d404b).

I can replicate this issue locally so am happy to do some testing for you when you make your change to see if the app builds.

**Comment by karlsc** [comment]
There you go @ethanneff 
```
https://github.com/phonegap-build/StatusBarPlugin.git   	git	*	0944be5	n/a	android,ios,winphone
branch-cordova-sdk   	npm	^2.5.0	2.6.2	n/a	android,ios,winphone
call-number   	npm	1.0.1	0.0.2	n/a	android,ios,winphone
https://github.com/Salesfloor/cordova-Pinterest.git   	git	*	e9bd76b	n/a	android,ios,winphone
cordova-custom-config   	npm	*	4.0.2	n/a	android,ios,winphone
cordova-instagram-plugin   	npm	0.5.3	0.5.3	n/a	android,ios,winphone
cordova-plugin-actionsheet   	npm	*	2.3.3	n/a	android,ios,winphone
cordova-plugin-buildinfo   	npm	1.1.0	1.1.0	n/a	android,ios,winphone
cordova-plugin-customurlscheme   	npm	*	4.3.0	n/a	android,ios,winphone
cordova-plugin-file   	npm	3.0.0	3.0.0	n/a	android,ios,winphone
cordova-plugin-file-transfer   	npm	1.4.0	1.4.0	n/a	android,ios,winphone
cordova-plugin-inappbrowser   	npm	1.1.1	1.1.1	n/a	android,ios,winphone
cordova-plugin-network-information   	npm	*	1.3.3	n/a	android,ios,winphone
cordova-plugin-splashscreen   	npm	*	4.0.3	n/a	android,ios,winphone
cordova-plugin-whitelist   	npm	1.3.1	1.3.1	n/a	android,ios,winphone
cordova-plugin-x-socialsharing   	npm	*	5.1.8	n/a	android,ios,winphone
org.apache.cordova.camera   	npm	2.3.1	2.3.1	n/a	android,ios,winphone
org.apache.cordova.contacts   	npm	2.2.0	2.2.0	n/a	android,ios,winphone
org.apache.cordova.device   	npm	1.1.3	1.1.3	n/a	android,ios,winphone
org.apache.cordova.dialogs   	npm	1.3.0	1.3.0	n/a	android,ios,winphone
org.apache.cordova.file   	npm	1.3.1	n/a	n/a	android,ios,winphone
org.apache.cordova.file-transfer   	npm	0.4.6	n/a	n/a	android,ios,winphone
org.apache.cordova.media   	npm	*	3.0.1	n/a	android,ios,winphone
org.apache.cordova.vibration   	npm	0.3.11	0.3.11	n/a	android,ios,winphone
phonegap-plugin-barcodescanner   	npm	6.0.5	6.0.5	n/a	android,ios,winphone
phonegap-plugin-push   	npm	1.9.4	1.9.4	n/a	android,ios,winphone
salesfloor-oauthio   	npm	*	0.2.6	n/a	android,ios,winphone
salesfloor-silent-mode-plugin   	npm	1.0.0	0.9.0	n/a	android,ios,winphone
```

**Comment by ethanneff** [comment]
Hello hello again, 

I pushed version `2.6.3` which removed the Google Play Services framework from our SDK.  We need to do some additional work and testing internally before we can confirm the upgrade from 9+ to 11+, so we have removed the framework for now. The good news, however, is this framework removal will not impact your deep linking abilities with Branch.

Thank you for `cordova plugins` output. I will be using them to investigate a better long term solution so our SDK does not impact other SDKs.

To can follow [this guide](https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking#compiling-updating-the-branch-sdk) to upgrade your Branch Cordova SDK to `2.6.3`. 

Thanks,
